### ISSUE TITLE
[Segnet] TypeError: new() received an invalid combination of arguments

### ISSUE THREAD

**Issue by tailongnguyen** [ZijunDeng/pytorch-semantic-segmentation#42]
Hi, 
I am using pytorch version 0.4.0, python 3.6 and this code appears to need a minor modification to make Segnet works. 
Here is the bug:
![image](https://user-images.githubusercontent.com/23494207/42810235-a8a2cbac-89e1-11e8-9974-ebb0e5e72624.png)
I found out that the line 12 should change from 
`middle_channels = in_channels / 2`
to
`middle_channels = in_channels // 2`

**Comment by oujieww** [comment]
@tailongnguyen is your training ok? i also want to train segnet on voc but with training go on ,my mean iu is always =0.03469,can you give me your setting about trainning code,i use the setting same as fcn,lr=2.5e-4

**Comment by tailongnguyen** [comment]
@oujieww 
I used this code but did not focused on this SegNet architecture, thus only tried on training with one set of hyper parameters (lr = 1e-4, negative log likelihood loss function with weighted scheme following E-Net) . The performance was unexpectedly bad, too. I am not sure if there was something wrong with the code or my dataset is just not suitable with this model, so I skipped it. 
However, if you ask about the training, my mean IoU DID change over the training phase.

**Comment by oujieww** [comment]
@tailongnguyen yeah, is your training mIoU is nromal?  such as bigger than 0.5?  and i have tried this code on fcn32s even is better than official code mIoU is 0.6447 i got. but for other model of this code i can not work,  ,XD , do u know where can find a good code of semantic segmentation , i want to do some test on segmentation  XD

**Comment by tailongnguyen** [comment]
Awesome-semantic-segmentation github repo is what you need.

**Comment by oujieww** [comment]
@tailongnguyen  thank you !!!

**Comment by robinreni96** [comment]
Its simple in the Decoderblock inside __init__ function instead of 
`middle_channels = in_channels / 2`
use
`middle_channels = int(in_channels / 2)`
